test/funder.js

```diff
@@ -0,0 +1,328 @@
+      }
+    });
+  };
+
+  if (Array.isArray(txHash) && txHash.length > 0) {
+    return Promise.all(txHash.map(
+      oneTxHash => self.getTransactionReceiptMined(oneTxHash, interval),
+    ));
+  }
+  if (typeof txHash === 'string') {
+    return new Promise(transactionReceiptAsync);
+  }
+  throw new Error(`Invalid Type: ${txHash}`);
+}
+
+/**
+ * This method captures detail about funding transaction from fauce like faucet
+ * address, eip20 token address etc. which is used to return remaining fund
+ * to faucet.
+ * @param {Array<string>} originFaucetTXHashes Transaction hashes of funding
+ *                                             transaction by mosaic faucet
+ *                                             on the origin chain.
+ * @param {Array<string>} auxiliaryFaucetTXHashes Transaction hashes of funding
+ *                                                transaction by mosaic faucet
+ *                                                on the auxiliary chain.
+ * @param {Array<string>} ropstenFaucetTXHashes Transaction hashes of funding
+ *                                              transaction by ropsten faucet
+ *                                              on the origin chain.
+ * @param {Web3} originWeb3 Web3 instance pointing to the origin chain.
+ * @param auxiliaryWeb3 Web3 instance pointing to the auxiliary chain.
+ * @return {Promise<RefundTransactionDetails>} Details of refund transaction
+ *                                            to mosaic and ropsten faucet.
+ */
+const faucetTransactionDetails = async (
+  originFaucetTXHashes,
+  auxiliaryFaucetTXHashes,
+  ropstenFaucetTXHashes,
+  originWeb3,
+  auxiliaryWeb3,
+) => {
+  // Get faucet address from one transaction.
+  const originTransaction = originWeb3.eth.getTransaction(originFaucetTXHashes[0]);
+  const auxiliaryTransaction = auxiliaryWeb3.eth.getTransaction(auxiliaryFaucetTXHashes[0]);
+  const ropstenTransaction = originWeb3.eth.getTransaction(ropstenFaucetTXHashes[0]);
+
+  return Promise.all(
+    [
+      originTransaction,
+      auxiliaryTransaction,
+      ropstenTransaction,
+    ],
+  ).then(transactions => ({
+    originTransactions: {
+      type: 'ERC20',
+      faucetAddress: transactions[0].from,
+      tokenAddress: transactions[0].to,
+
+    },
+    auxiliaryTransactions: {
+      type: 'BaseToken',
+      faucetAddress: transactions[1].from,
+    },
+    ropstenTransactions: {
+      type: 'BaseToken',
+      faucetAddress: ROPSTEN_REFUND_ADDRESS,
+    },
+  }));
+};
+
+/**
+ * This method returns all the base token balance to faucet address from the
+ * `from` address.
+ * @param {Web3} web3 Instance of web3.
+ * @param {string} from Address from which fund will be transferred.
+ * @param {string} faucetAddress Address of the faucet.
+ */
+const refundBaseTokenToFaucet = async (web3, from, faucetAddress) => {
+  const balance = new BN(await web3.eth.getBalance(from));
+  const gasPrice = new BN(await web3.eth.getGasPrice());
```

Comment:
If you get the gas price here, should you not also set it on the transaction options to ensure that the remaining balance equals the value that you want to transfer?